### PR TITLE
OCPCLOUD-2493: Add unit tests for the functions handling ScaleFromZero annotations

### PR DESCRIPTION
**Pull request by racheljpg**
Hello, this is a PR to add unit tests to the MAO to test the scale from zero functions. Thanks!

### CODE REVIEW

File: pkg/util/machineset/util_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+			target.SetAnnotations(tc.expectedAnnotations)
+			err := target.UpdateScaleFromZeroAnnotations()
+			if err != nil {
+				t.Errorf("Unexpected error parsing ScaleFromZero annotations :%v", err)
+			}
+
+			observed := target.ParseMachineSetAnnotationKey()
+
+			if observed.expectedReturn == false {
+				target.Expect(err).ToNot(HaveOccured())
```
Comment (elmiko) [inline pkg/util/machineset/util_test.go:100]:
```suggestion
				Expect(err).ToNot(HaveOccured())
```

File: pkg/util/machineset/util_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+			} else {
+				target.Expect(err).To(HaveOccured())
```
Comment (elmiko) [inline pkg/util/machineset/util_test.go:102]:
```suggestion
				Expect(err).To(HaveOccured())
```

File: pkg/util/machineset/util_test.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+			expectedReturn: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			target := NewWithT(t)
+			target.SetAnnotations(tc.expectedAnnotations)
+			err := target.UpdateScaleFromZeroAnnotations()
+			if err != nil {
```
Comment (elmiko) [inline pkg/util/machineset/util_test.go:93]: probably should remove this or the test will stop whenever we get a failure here

File: pkg/util/machineset/util_test.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			target := NewWithT(t)
+			target.SetAnnotations(tc.expectedAnnotations)
+			err := target.UpdateScaleFromZeroAnnotations()
+			if err != nil {
```
Comment (elmiko) [inline pkg/util/machineset/util_test.go:145]: same here, if we want to check the error later we can't stop here

File: pkg/util/machineset/util_test.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+				t.Errorf("Unexpected error checking ScaleFromZero annotations :%v", err)
+			}
+
+			observed := target.HasScaleFromZeroAnnotationsEnabled()
+
+			if observed.expectedReturn == false {
+				target.Expect(err).ToNot(HaveOccured())
```
Comment (elmiko) [inline pkg/util/machineset/util_test.go:152]:
```suggestion
				Expect(err).ToNot(HaveOccured())
```

File: pkg/util/machineset/util_test.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+			} else {
+				target.Expect(err).To(HaveOccured())
```
Comment (elmiko) [inline pkg/util/machineset/util_test.go:154]:
```suggestion
				Expect(err).To(HaveOccured())
```

**Review by elmiko — COMMENTED** [review]
i think you just had some of the ginkgo call setup wrong

**Comment by elmiko** [comment]
looks like we need to use `NewWithT` to get the gomega/ginkgo stuff configured properly. i tried [this out](https://github.com/elmiko/machine-api-operator/commit/70a925bc02abb4615db2cd578557f12a373633a7) and it worked (although there is an error in the tests for you to solve XD).

**Comment by racheljpg** [comment]
@elmiko thanks Mike! all solved now :) lmk if/when tests pass if you think this is good to merge? thanks!

File: pkg/util/machineset/util_test.go [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+func TestSettingAnnotations(t *testing.T) {
+	tests := []struct {
+		name                string
+		value               string
+		fn                  func(map[string]string, string) map[string]string
+		expectedAnnotations map[string]string
+		suppliedAnnotations map[string]string
+	}{
+		{
```
Comment (elmiko) [inline pkg/util/machineset/util_test.go:18]: let's make sure we have a test case that sets an empty string so that we know it works

**Review by elmiko — COMMENTED** [review]
tests are looking good, just one suggestion

also, i think we are fine to take this out of draft

File: pkg/util/machineset/util_test.go (hunk 7 above)
Comment (racheljpg) [reply pkg/util/machineset/util_test.go:18]: will do!

**Review by elmiko — COMMENTED** [review]
thanks for updating

/lgtm

**Review by damdo — COMMENTED** [review]
/lgtm

Thanks!
